HEC.FDA.Model/paireddata/CurveMetaData.cs

```diff
@@ -63,69 +51,76 @@ public CurveMetaData(string xlabel, string ylabel, string name, string damageCat
+            ImpactAreaID = impactAreaID;
         }
         public XElement WriteToXML()
         {
-            XElement masterElement = new XElement("Curve_Metadata");
-            masterElement.SetAttributeValue("CurveType", Convert.ToString(CurveType));
-            masterElement.SetAttributeValue("XLabel", XLabel);
-            masterElement.SetAttributeValue("YLabel", YLabel);
-            masterElement.SetAttributeValue("Name", Name);
-            masterElement.SetAttributeValue("DamageCategory", DamageCategory);
-            masterElement.SetAttributeValue("AssetCategory", AssetCategory);
-            masterElement.SetAttributeValue("IsNull", IsNull);
+            StoredPropertyAttribute attribute = (StoredPropertyAttribute)Attribute.GetCustomAttribute(typeof(CurveMetaData), typeof(StoredPropertyAttribute));
+            string thresholdMasterTag = attribute.SerializedName;
+
+            XElement masterElement = new XElement(thresholdMasterTag);
+
+            string xLabelTag = Serialization.GetXMLTagFromProperty(GetType(), nameof(XLabel));
+            masterElement.SetAttributeValue(xLabelTag, XLabel);
+
+            string yLabelTag = Serialization.GetXMLTagFromProperty(GetType(),nameof(YLabel));
+            masterElement.SetAttributeValue(yLabelTag, YLabel);
+
+            string nameTag = Serialization.GetXMLTagFromProperty(GetType(),nameof(Name));
+            masterElement.SetAttributeValue(nameTag, Name);
+
+            string damCatTag = Serialization.GetXMLTagFromProperty(GetType(),nameof(DamageCategory));
+            masterElement.SetAttributeValue(damCatTag, DamageCategory);
+
+            string assetCatTag = Serialization.GetXMLTagFromProperty(GetType(),nameof(AssetCategory));
+            masterElement.SetAttributeValue(assetCatTag, AssetCategory);
+
+            string isNullTag = Serialization.GetXMLTagFromProperty(GetType(),nameof(IsNull));
+            masterElement.SetAttributeValue(isNullTag, IsNull);
             return masterElement;
         }
 
         public static CurveMetaData ReadFromXML(XElement xElement)
         {
-            CurveTypesEnum curveType = (CurveTypesEnum)Enum.Parse(typeof(CurveTypesEnum), xElement.Attribute("CurveType").Value);
-            string xLabel = xElement.Attribute("XLabel").Value;
-            string yLabel = xElement.Attribute("YLabel").Value;
-            string name = xElement.Attribute("Name").Value;
-            string damageCategory = xElement.Attribute("DamageCategory").Value;
-            string assetCategory = xElement.Attribute("AssetCategory").Value;
-            CurveMetaData curveMetaData = new CurveMetaData(xLabel, yLabel, name, damageCategory, curveType, assetCategory);
-            bool isNull = Convert.ToBoolean(xElement.Attribute("IsNull").Value);
+            CurveMetaData metaDataForReflection = new();
+
+            string xLabelTag = Serialization.GetXMLTagFromProperty(metaDataForReflection.GetType(), nameof(XLabel));
```

Comment:
Don't get this type over and over again. Instead of making the empty Curve Meta Data, do something like

` Type metaDataType = typeof(CurveMetaData);

Then just pass in metaDataType

Reply:
done